### PR TITLE
483 - Unsaved changes modal

### PR DESCRIPTION
**Pull request by Kasmadei**
Point for: https://github.com/kbss-cvut/fta-fmea-ui/issues/483
(if FTA is deselected user is notified by a modal and can SAVE/DISCARD changes)
Has eng/cz text.
Screenshot 2024-07-17 at 3 18 33 PM

### CODE REVIEW

File: src/components/editor/faultTree/canvas/EditorCanvas.tsx [hunk 1]
```diff
@@ -347,7 +347,7 @@ const EditorCanvas = ({
         </CurrentFaultTreeTableProvider>
         {!showTable && (
           <FaultEventMenu
-            shapeToolData={sidebarSelectedEvent}
+            newShapeToolData={sidebarSelectedEvent}
```
Comment (blcham) [inline src/components/editor/faultTree/canvas/EditorCanvas.tsx:350]: Why newShapeToolData ?

**Review by blcham — CHANGES_REQUESTED** [review]

**Comment by blcham** [comment]
We are still not showing modal for:
- diagram options 
- back button

File: src/components/editor/faultTree/canvas/EditorCanvas.tsx (hunk 1 above)
Comment (blcham) [reply src/components/editor/faultTree/canvas/EditorCanvas.tsx:350]: selectedShapeToolData

**Comment by blcham** [comment]
@LaChope please have a look so you know what is @Kasmadei implementing

**Review by LaChope — APPROVED** [review]
Looks good to me, LGTM after requested changes by @blcham were implemented.

**Pull request merged**